index.js

```diff
@@ -0,0 +1,114 @@
+import * as core from "@actions/core";
+import fetch from "node-fetch";
+
+const NETLIFY_BASE_URL = `https://api.netlify.com/api/v1/`;
+const READY_STATES = ["ready", "current"];
+const WAIT_TIMEOUT = 60 * 15; // 15 min
+const WAIT_INCREMENT = 15; // seconds
+
+const netlifyFetch = async (netlifyToken, url) => {
+  return await fetch(NETLIFY_BASE_URL + url, {
+    headers: {
+      Authorization: `Bearer ${netlifyToken}`,
+    },
+  }).then((response) => response.json());
+};
+
+const getDeploy = async (inputs) => {
+  const deploys = await netlifyFetch(
+    inputs.netlifyToken,
+    `sites/${inputs.siteId}/deploys?branch=${inputs.branch}`
+  );
+
+  return deploys.find(
+    (d) => d.context == "production" && d.commit_ref == inputs.commitSha
+  );
+};
+
+const isDeployReady = (deploy) => {
+  return READY_STATES.includes(deploy.state);
+};
+
+const waitForDeployToBeReady = async (inputs, deploy) => {
+  let waitTime = 0;
+
+  core.info(
+    `Production deploy for branch '${inputs.branch}' (sha: ${inputs.commitSha}) ` +
+      `is not ready yet.\nWaiting ${WAIT_INCREMENT} more seconds for the ` +
+      `deploy to finish.`
+  );
+
+  const handle = setInterval(
+    async (inputs, deployId) => {
+      waitTime += WAIT_INCREMENT;
+
+      if (waitTime >= WAIT_TIMEOUT) {
+        clearInterval(handle);
```

Comment:
wait, this works even though we're inside of the interval function? :raised_eyebrow:

Reply:
Yup.